packages/sdk/src/evm/core/sdk.ts

```diff
@@ -90,12 +91,18 @@ export class ThirdwebSDK extends RPCConnectionHandler {
    */
   static fromSigner(
     signer: Signer,
-    network?: ChainIdOrNameOrChain,
+    network?: ChainOrRpcUrl,
     options: SDKOptions = {},
     storage: ThirdwebStorage = new ThirdwebStorage(),
   ): ThirdwebSDK {
-    const sdk = new ThirdwebSDK(network || signer, options, storage);
-    sdk.updateSignerOrProvider(signer);
+    // If we are given a network, connect the signer to that network
+    let signerWithProvider = signer;
+    if (network) {
+      const provider = getChainProvider(network, options);
+      signerWithProvider = signer.connect(provider);
```

Comment:
This can throw for certain signers. We should also check if there's already a provider set, otherwise we might overriding custom providers here

Reply:
If they specify a network + there's already a provider connected to the signer + the provider is on the wrong network, we should probably still overwrite. Will update for that case.

Reply:
oof, that would make it async hmm

Reply:
In that case, assuming we cant make `fromSigner` async, I actually prefer always setting provider ourselves then if they specify network. Network is optional here, so if they don't pass it then I think of it as "use the provider on my signer". But if they do pass it, I think of it as "connect my signer to this new provider I'm telling you to."

Ex: they can pass an RPC URL to the network argument, and it would be very weird if we just ignored it and didn't connect their signer to that RPC URL just because it already had a provider - wdyt?

Reply:
If you connect a random provider to a signer that can throw for some signers if the provider is not on the same chain as the signer

Reply:
I think this should throw an error in this case

Reply:
might be ok because network is optional. guess we can try it

Reply:
adding the `!signer.provider` seems fine tho